packages/core/src/amazonq/commons/controllers/contentController.ts

```
@@ -52,4 +69,94 @@ export class EditorContentController {
+     */
+    public async acceptDiff(message: any) {
+        const { filePath, selection } = extractFileAndCodeSelectionFromMessage(message)
+
+        if (filePath && message?.code?.trim().length > 0 && selection) {
+            try {
+                const doc = await vscode.workspace.openTextDocument(filePath)
+
+                const code = getIndentedCode(message, doc, selection)
+                const range = getSelectionFromRange(doc, selection)
+                try {
+                    await applyChanges(doc, range, code)
+
+                    // Sets the editor selection from the start of the given range, extending it by the number of lines in the code till the end of the last line
+                    const editor = await vscode.window.showTextDocument(doc)
+                    editor.selection = new vscode.Selection(
+                        range.start,
+                        new Position(range.start.line + code.split('\n').length, Number.MAX_SAFE_INTEGER)
+                    )
+                } catch (error) {
+                    getLogger().error(`Diff: Unable to apply changes in the document:`, (error as Error).message)
+                }
+
+                // If vscode.diff is open for the filePath then close it.
+                vscode.window.tabGroups.all.flatMap(({ tabs }) =>
+                    tabs.map((tab) => {
+                        if (tab.label === `${path.basename(filePath)} ${amazonQTabSuffix}`) {
+                            const tabClosed = vscode.window.tabGroups.close(tab)
+                            if (!tabClosed) {
+                                getLogger().error('Diff: Unable to close the diff view tab for %s', tab.label)
+                            }
+                        }
+                    })
+                )
+            } catch (error) {
+                getLogger().error(`Diff: Unable to open the document:`, (error as Error).message)
```

What is the behavior if we were to throw instead? If this code is initially triggered in a Command, then the error will be swallowed upstream, and logged.

---

Discussed offline. 

Display an error Notification for the user. Update the logic to throw everywhere and catch at high level and display a generic error notification.